homeworks/hw12/src/bank.rs

```diff
@@ -0,0 +1,869 @@
+                    amount,
+                    operation_type: OperationType::Deposit,
+                };
+                // self.history.push_back(operation);
+                self.push_transaction(operation);
+                info!("Deposited into account {}", &account);
+                Ok(transaction_id.to_owned())
+            }
+        } else {
+            error!("Account does not exist");
+            Err(AccountNotFoundError {
+                account: account.to_owned(),
+            }
+            .into())
+        }
+    }
+    /// Withdraws the specified amount from the account.
+    ///
+    /// # Arguments
+    ///
+    /// * `amount` - The amount to withdraw from the account.
+    /// * `account` - The account to withdraw from.
+    ///
+    /// # Errors
+    /// AmountNegativeError
+    /// AccountNotFoundError
+    /// InsufficientFundsError
+    ///
+    /// Returns an error if the account balance is insufficient to cover the withdrawal amount.
+    ///
+    /// ```
+
+    fn withdraw(&mut self, account: &str, amount: Money) -> Result<TransactionId, BankError> {
+        let transaction_id = get_next_id();
+        if let Some(balance) = self.accounts.get_mut(account) {
+            if amount <= Money::default() {
+                error!("Amount must be positive: amount {amount}");
+                Err(AmountNegativeError {
+                    account: account.to_owned(),
+                    amount,
+                }
+                .into())
+            } else if *balance < RefCell::from(amount) {
+                let balance = balance.borrow();
+                error!(
+                    "Insufficient funds for the operation. Balance: {balance:?} Amount: {amount}"
+                );
+                Err(InsufficientFundsError {
+                    amount,
+                    account: account.to_owned(),
+                    balance: balance.to_owned(),
+                }
+                .into())
+            } else {
+                let mut balance = balance.borrow_mut();
+                debug!("Balance before: {balance:?}");
+                *balance -= amount;
+
+                let operation = Operation {
+                    id: transaction_id.to_owned(),
+                    source_account: account.to_owned(),
+                    target_account: None,
+                    amount,
+                    operation_type: OperationType::Withdraw,
+                };
+                // self.history.push_back(operation);
+
+                if self.accounts_history.contains_key(account) {
+                    self.accounts_history
+                        .get_mut(account)
+                        .unwrap()
+                        .push(operation.id.clone());
+                } else {
+                    self.accounts_history
+                        .insert(account.to_owned(), vec![operation.id.clone()]);
+                }
+                self.history.insert(operation.id.clone(), operation);
```

Comment:
```suggestion
self.push_transaction(operation)
```

Вроде должно быть так

Reply:
Да, была трабла с  borrow-checker - 2 мутабельные ссылки. Разнес операции.